File: HW2/Dellal_Zeynep_Doga.rb
puts "--UNCONDITIONAL BREAK--";
i  = 1;
for i in 1..5 do
  puts i;
  i = i + 1;
  break;
end

puts "--UNCONDITIONAL CONTINUE--";
i  = 1;
for i in 1..5 do
  puts i;
  i = i + 1;
  next;
end

puts "--UNLABELED BREAK--";
i  = 1;
a = 1;
while i < 5 do
    if i == 4 then break end
    puts i;
    i = i+1;
  end

puts "--UNLABELED CONTINUE--";
i  = 0;
a = 0;
while i < 5 do
    i = i+1;
    if i == 3 then next end
    puts i;
  end

File: HW2/Dellal_Zeynep_Doga.php
<?php
echo "--UNCONDITIONAL BREAK-- <br>";
for ($i = 1; $i < 5; $i++) {
  echo "i: $i <br>";
  break;
}

echo "--UNCONDITIONAL BREAK-- <br>";
for($i = 1; $i < 5; $i++) {
  echo "i: $i <br>";
  continue;
}

$i = 1;
$a = 1;
echo "--UNLABELED BREAK-- <br>";
while($i < 5) {
  echo "i: $i <br>";
  $i++;
  if ($i == 4) break;
}
$i = 0;
echo "--UNLABELED CONTINUE-- <br>";
while($i < 5) {
  $i++;
  if ($i == 3) continue;
  echo "i: $i <br>";
}

$i = 0;
$a = 0;
echo "--BREAK FROM OUTER LOOP-- <br>";
while($i < 5) {
 echo "";
    while($a < 5){
      $a++;
      $i++;
      if ($a == 3) {
          break 2;
      }
      echo "i in outer loop: $i , a in inner loop: $a <br>";
    }
  }
  
$i = 0;
$a = 0;
echo "--CONTINUE FROM OUTER LOOP-- <br>";
while($i < 5) {
    while($a < 5){
      $a++;
      $i++;
      if ($a == 3) {
          continue 2;
      }
      echo "i in outer loop: $i , a in inner loop: $a <br>";
    }
  }
?>